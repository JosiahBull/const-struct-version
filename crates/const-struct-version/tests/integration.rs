use const_struct_version::StructVersion;

#[allow(dead_code)]
#[test]
fn test_that_different_name_same_hash() {
    #[derive(StructVersion)]
    struct Cart1 {
        value: i32,
    }

    #[derive(StructVersion)]
    struct Cart2 {
        value: i32,
    }

    let version1 = <Cart1 as StructVersion>::version();
    let version2 = <Cart2 as StructVersion>::version();

    assert_eq!(version1, version2);
}
