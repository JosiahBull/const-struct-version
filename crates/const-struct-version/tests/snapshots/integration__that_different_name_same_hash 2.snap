---
source: crates/const-struct-version/tests/integration.rs
expression: version1
---
fba45310542184ee6005afa7ce9e2c4d74dd66ed
